crates/revm/src/db/ethersdb.rs

```diff
@@ -40,9 +33,13 @@ impl<M: Middleware> EthersDB<M> {
 
     /// internal utility function to call tokio feature and wait for output
     fn block_on<F: core::future::Future>(&self, f: F) -> F::Output {
-        match &self.runtime {
-            Some(runtime) => runtime.block_on(f),
-            None => futures::executor::block_on(f),
+        match Handle::try_current() {
+            Ok(handle) => handle.block_on(f),
```

Comment:
this will still panic if called from the main Runtime::block_on thread.

what we want here is

```rust
tokio::task::block_in_place(move || {
    handle.block_on(f)
});
```

Reply:
Yes, I just noticed the problem. Actually, I think it looks impossible to run a future in a non-async function within a tokio runtime (?.

Reply:
@mattsse

I investigated a bit. It looks like we can't support `current_thread` due to not being able to `blocking_in_place` without changing the signature of the `Database(Ref)` traits. I have two options here:

1. Document clearly that, `EthersDB` can't be used with `current_thread`.
2. Less preferably, we start another thread and create another runtime to do it.


Which one do you prefer?

Reply:
Done with 2 so we can support all use cases, could you have a look? @mattsse

Reply:
There is issue related to this, not sure if you have looked at it: https://github.com/bluealloy/revm/issues/1056

Reply:
Yes, I tested this addresses #1056 cleanly.